File: UI/careconnect_mobile/lib/models/responses/payment_intent_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'payment_intent_response.g.dart';

@JsonSerializable()
class PaymentIntentResponse {
  String clientSecret;
  String paymentIntentId;
  double amount;
  String currency;

  PaymentIntentResponse({
    required this.clientSecret,
    required this.paymentIntentId,
    required this.amount,
    required this.currency,
  });

  factory PaymentIntentResponse.fromJson(Map<String, dynamic> json) =>
      _$PaymentIntentResponseFromJson(json);

  Map<String, dynamic> toJson() => _$PaymentIntentResponseToJson(this);
}

File: CareConnect/CareConnect.Models/Responses/PaymentIntentResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CareConnect.Models.Responses
{
    public class PaymentIntentResponse
    {
        public string ClientSecret { get; set; } = null!;

        public string PaymentIntentId { get; set; } = null!;

        public decimal Amount { get; set; }

        public string Currency { get; set; } = null!;
    }
}
